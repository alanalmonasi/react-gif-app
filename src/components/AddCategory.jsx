import PropTypes from 'prop-types'
import { useState } from "react"


export const AddCategory = ({ onNewValue }) => {

   const [inputValue, setInputValue] = useState('');

   const onInputChange = (e) => {
      setInputValue(e.target.value);
   }
   
   const onSubmit = (e) => {
      e.preventDefault();

      if(inputValue.length <= 1) return; 
      
      setInputValue('');

      onNewValue(inputValue.trim().toLocaleUpperCase());
   }

   return (
      <form onSubmit={onSubmit} aria-label="form">

         <input 
            type="text"
            placeholder="Buscar gif"
            value={inputValue}
            onChange={onInputChange}
         />
      </form>
   )
}

AddCategory.propTypes = {
   onNewValue: PropTypes.func.isRequired,
}