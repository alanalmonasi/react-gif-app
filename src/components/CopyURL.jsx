import { useState } from "react";

export const CopyURL = ({ url }) => {
   const [copied, setCopied] = useState(false);
   const link = url.split('?')

   const copy = () => { 
      const el = document.createElement('input');
      el.value = link[0];
      document.body.appendChild(el);
      el.select();
      document.execCommand('copy');
      document.body.removeChild(el);
      setCopied(true);
   }
   
   return (
      <div className="copy-link">
         <button onClick={copy}>{!copied ? "Copy link" : "Copied!"}</button>
      </div>
   );
}


