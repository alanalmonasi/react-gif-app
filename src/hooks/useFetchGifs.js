import { useState, useEffect } from "react";
import { getGifs } from '../helpers/getGifs'

export const useFetchGifs = (category) => {

   const [images, setImage] = useState([]);
   const [isLoading, setIsLoading] = useState(true);

   const getImages = async () => {
      const newImage = await getGifs(category);
      setImage(newImage);
      setIsLoading(false);
   }

   useEffect(() => {
      getImages();
   }, []);

   return{
      images,
      isLoading
   }
}
