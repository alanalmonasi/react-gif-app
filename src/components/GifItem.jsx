import PropTypes from 'prop-types'

import React from 'react'
import { CopyURL } from './CopyURL'

export const GifItem = ({ title, url }) => {
  return (
      <div className="card">
         <img src={url} alt={title} />
         <p>{title}</p>
         <CopyURL url={url}/>
      </div>
   )
}

GifItem.propTypes = {
   title: PropTypes.string.isRequired,
   url: PropTypes.string.isRequired
}
