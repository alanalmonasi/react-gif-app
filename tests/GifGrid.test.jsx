import { render, screen } from '@testing-library/react';
import { GifGrid } from '../src/components/GifGrid'
import { useFetchGifs } from '../src/hooks/useFetchGifs';

jest.mock('../src/hooks/useFetchGifs');

describe('testing GifGrid component', () => { 

   const category = 'GOKU';
   
   test('should show loading', () => { 
      
      useFetchGifs.mockReturnValue({
         images: [],
         isLoading: true,
      });

      render( <GifGrid category={category}/>)
      screen.debug();

      expect(screen.getByText('Cargando gifs...'));
      expect(screen.getByText(category)).toBeTruthy();
   });

   test('should show items when use useFetchGifs', () => { 

      const gifs = [
         {
            id: 'abc',
            title: 'goku',
            url: 'https://www.google.com/'
         },
         {
            id: 'abcd',
            title: 'krilin',
            url: 'https://www.google.com/'
         }
   ];

      useFetchGifs.mockReturnValue({
         images: gifs,
         isLoading: false,
      });
      
      const { container } = render(<GifGrid category={category}/>);
      expect(screen.getAllByRole('img').length).toBe(2);
      expect(container).toMatchSnapshot();
   });

});