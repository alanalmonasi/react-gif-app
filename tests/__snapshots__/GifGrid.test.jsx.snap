// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing GifGrid component should show items when use useFetchGifs 1`] = `
<div>
  <h3>
    GOKU
  </h3>
  <div
    class="card-grid"
  >
    <div
      class="card"
    >
      <img
        alt="goku"
        src="https://www.google.com/"
      />
      <p>
        goku
      </p>
      <div
        class="copy-link"
      >
        <button>
          Copy link
        </button>
      </div>
    </div>
    <div
      class="card"
    >
      <img
        alt="krilin"
        src="https://www.google.com/"
      />
      <p>
        krilin
      </p>
      <div
        class="copy-link"
      >
        <button>
          Copy link
        </button>
      </div>
    </div>
  </div>
</div>
`;
